### PR TITLE
Added Python DFS code to Silver Graph Traversals

### PR DESCRIPTION
**Pull request by Genius3435**
Main changes
- DFS solutions work in Python when submitted in CPython3 (PyPy3 doesn't even pass with FastIO; I assume the JIT compiler is doing something suboptimal on that one testcase)
- Changed `visited` to `vis` for consistency

_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [X] I have tested my code.
- [X] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [X] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

File: content/3_Silver/Graph_Traversal.mdx [hunk 1]
```diff
@@ -552,19 +528,19 @@ using namespace std;
 
 int N;
 vector<vector<int>> adj;
-vector<bool> visited;
+vector<bool> vis;
```
Comment (SansPapyrus683) [inline content/3_Silver/Graph_Traversal.mdx:531]: bruh.
change this back

File: content/3_Silver/Graph_Traversal.mdx [hunk 2]
```diff
@@ -652,35 +628,36 @@ public class Main {
...
+for _ in range(m):
+	u, v = map(int, input().split())
+	adj[u].append(v)
+	adj[v].append(u)
+
+vis = [False] * (n + 1)
+rep = []
+
+
+def dfs(node):
```
Comment (SansPapyrus683) [inline content/3_Silver/Graph_Traversal.mdx:642]: type annotations + make the structure consistent w/ other langs

File: content/3_Silver/Graph_Traversal.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply content/3_Silver/Graph_Traversal.mdx:531]: (everywhere else too)

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/3_Silver/Graph_Traversal.mdx [hunk 3]
```diff
@@ -771,42 +746,44 @@ This will occur for $N>10^3$ since the recursion limit in Python is set to
...
+
+
+def count_components():
+	for i in range(1, n + 1):
+		if not vis[i]:
+			rep.append(i)
+			dfs(i)
+
+
+count_components()
```
Comment (ryanchou-dev) [inline content/3_Silver/Graph_Traversal.mdx:777]:
```suggestion
rep = []

def dfs(node):
	vis[node] = True
	for u in adj[node]:
		if not vis[u]:
			dfs(u)

def count_components():
	for i in range(1, n + 1):
		if not vis[i]:
			rep.append(i)
			dfs(i)

count_components()
```

File: content/3_Silver/Graph_Traversal.mdx [hunk 4]
```diff
@@ -1046,6 +1023,42 @@ public class BuildingTeams {
...
+	color[node] = col
+	for u in adj[node]:
+		if color[u] == -1:
+			dfs(u, col ^ 3)
+		elif color[u] == col:
+			print("IMPOSSIBLE")
+			exit(0)
+
+
+for u in range(n):
```
Comment (ryanchou-dev) [inline content/3_Silver/Graph_Traversal.mdx:1054]:
```suggestion
color = [-1] * n

def dfs(node, col):
	color[node] = col
	for u in adj[node]:
		if color[u] == -1:
			dfs(u, col ^ 3)
		elif color[u] == col:
			print("IMPOSSIBLE")
			exit(0)

for u in range(n):
```

File: content/3_Silver/Graph_Traversal.mdx [hunk 5]
```diff
@@ -771,42 +746,44 @@ This will occur for $N>10^3$ since the recursion limit in Python is set to
 ```py
 import sys
 
-sys.setrecursionlimit(1000000)
+sys.setrecursionlimit(10**6)
 
 # BeginCodeSnip{rest of solution}
```
Comment (ryanchou-dev) [inline content/3_Silver/Graph_Traversal.mdx:751]:
```suggestion
# BeginCodeSnip{Rest of solution}
```

**Pull request closed without merging**